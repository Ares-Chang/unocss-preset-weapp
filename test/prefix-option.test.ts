import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'
import presetWeapp from '../src/index'

describe('prefix', () => {
  test('preset prefix', async () => {
    const uno = createGenerator({
      presets: [
        presetWeapp({ prefix: 'h-' }),
      ],
      rules: [
        ['bar', { color: 'bar' }, { prefix: 'bar-' }],
      ],
      shortcuts: [
        ['shortcut', 'bar-bar', { prefix: 'bar-' }],
      ],
    })

    const unexpected = [
      'text-red',
      'hover:p4',
      'bar',
      'shortcut',
    ]

    const expected = [
      'h-text-red',
      'hover:h-p4',
      'bar-bar',
      'bar-shortcut',
      'h-space-x-4',
      'h-divide-x',
    ]

    const { css, matched } = await uno.generate(new Set([
      ...unexpected,
      ...expected,
    ]), { preflights: false })

    expect([...matched].sort()).toEqual(expected.sort())
    expect(css).toMatchSnapshot()
  })

  test('multiple preset prefix', async () => {
    const uno = createGenerator({
      presets: [
        presetWeapp({ prefix: ['h-', 'x-'] }),
      ],
      rules: [
        ['bar', { color: 'bar' }, { prefix: ['bar-', 'foo-', ''] }],
        [/^regex$/, () => ({ color: 'regex' }), { prefix: ['bar-', 'foo-', ''] }],
      ],
      shortcuts: [
        ['shortcut', 'bar-bar', { prefix: ['bar-', 'x-'] }],
      ],
    })

    const unexpected = [
      'text-red',
      'hover:p4',
      'bar',
      'shortcut',
    ]

    const expected = [
      'h-text-red',
      'hover:h-p4',

      'bar-bar',
      'foo-bar',
      'bar',

      'bar-regex',
      'foo-regex',
      'regex',

      'bar-shortcut',
      'x-shortcut',

    ]

    const { css, matched } = await uno.generate(new Set([
      ...unexpected,
      ...expected,
    ]), { preflights: false })

    expect([...matched].sort()).toEqual(expected.sort())
    expect(css).toMatchSnapshot()
  })
})
