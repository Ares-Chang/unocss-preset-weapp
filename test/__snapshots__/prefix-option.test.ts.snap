// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prefix > multiple preset prefix 1`] = `
"/* layer: default */
.hover_cl_h-p4:hover{padding:32rpx;}
.h-text-red{--un-text-opacity:1;color:rgba(248,113,113,var(--un-text-opacity));}
.bar,
.bar-bar,
.bar-shortcut,
.foo-bar,
.x-shortcut{color:bar;}
.bar-regex,
.foo-regex,
.regex{color:regex;}"
`;

exports[`prefix > preset prefix 1`] = `
"/* layer: default */
.h-divide-x>view+view,.h-divide-x>button+button,.h-divide-x>text+text,.h-divide-x>image+image,.h-divide-x>uni-view+uni-view,.h-divide-x>uni-button+uni-button,.h-divide-x>uni-text+uni-text,.h-divide-x>uni-image+uni-image,.h-divide-x>taro-view-core+taro-view-core,.h-divide-x>taro-image-core+taro-image-core,.h-divide-x>taro-text-core+taro-text-core,.h-divide-x>taro-button-core+taro-button-core{--un-divide-x-reverse:0;border-left-width:calc(1px * calc(1 - var(--un-divide-x-reverse)));border-right-width:calc(1px * var(--un-divide-x-reverse));border-left-style:solid;border-right-style:solid;}
.hover_cl_h-p4:hover{padding:32rpx;}
.h-text-red{--un-text-opacity:1;color:rgba(248,113,113,var(--un-text-opacity));}
.h-space-x-4>view+view,.h-space-x-4>button+button,.h-space-x-4>text+text,.h-space-x-4>image+image,.h-space-x-4>uni-view+uni-view,.h-space-x-4>uni-button+uni-button,.h-space-x-4>uni-text+uni-text,.h-space-x-4>uni-image+uni-image,.h-space-x-4>taro-view-core+taro-view-core,.h-space-x-4>taro-image-core+taro-image-core,.h-space-x-4>taro-text-core+taro-text-core,.h-space-x-4>taro-button-core+taro-button-core{--un-space-x-reverse:0;margin-left:calc(32rpx * calc(1 - var(--un-space-x-reverse)));margin-right:calc(32rpx * var(--un-space-x-reverse));}
.bar-bar,
.bar-shortcut{color:bar;}"
`;
